runtime/common/src/test/kotlin/kotlinx/serialization/features/ContextAndPolymorphicTest.kt

```diff
@@ -49,54 +47,59 @@ class CustomSerializersJvmTest {
         override val descriptor: SerialDescriptor = SerialClassDescImpl("Payload")
 
         override fun serialize(encoder: Encoder, obj: Payload) {
-            encoder.encodeString(HexConverter.printHexBinary(obj.s.toByteArray()))
+            encoder.encodeString(HexConverter.printHexBinary(obj.s.toUtf8Bytes()))
         }
 
         override fun deserialize(decoder: Decoder): Payload {
-            return Payload(String(HexConverter.parseHexBinary(decoder.decodeString())))
+            return Payload(stringFromUtf8Bytes(HexConverter.parseHexBinary(decoder.decodeString())))
         }
     }
 
     private val obj = EnhancedData(Data(100500), Payload("string"), Payload("binary"))
     private lateinit var json: Json
 
-    @Before
+    @BeforeTest
     fun initContext() {
-        val scope = SimpleModule(Payload::class, PayloadSerializer)
-        json = Json(unquoted = true).apply { install(scope) }
+        val scope = SingletonModule(Payload::class, PayloadSerializer)
+        val bPolymorphicModule = object : SerialModule {
```

Comment:
This pattern is already frequent in tests.
It would be nice to have something like
`fun PolymorphicModule(baseClass, actualClass, serializer)`, maybe with reified overloads

Reply:
I think the approach in https://github.com/Kotlin/kotlinx.serialization/pull/268/files/cfce6687f9828d0d7425f461d338199198e6b565?utf8=%E2%9C%93&diff=unified#diff-f4c790c741d52624028aea493fb2a532R49 is quite short and I can rewrite this place to use it. Usually, people want to register more than one subclass serializer.